flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/source/reader/IcebergSourceSplitReader.java

```diff
@@ -80,6 +88,35 @@ public RecordsWithSplitIds<RecordAndPosition<T>> fetch() throws IOException {
     }
 
     if (currentReader.hasNext()) {
+      if (pausedSplits.contains(currentSplitId)) {
```

Comment:
I would like to avoid the whole logic in the hot path. This is the reader where every Ms should be shaved off if possible.

Reply:
not sure I got the performance difference. we are talking about two options

option 1
```
if (condition) {
    ...
    while (condition) {
        ...
    }
}
```

option 2
```
while (condition) {
    ...
}
```

Reply:
The current code adds logging as well, so we can understand why the thread is paused.
OTOH we do not want to write one log line in every loop (every second). For this we need to add a variable (named `first` in the code). This variable creation is inside of the `if` codepath to prevent unnecessary variable creation in the happy path. That is the only point where the option 1 and option 2 is different.